docs/design/configs.md

```diff
@@ -0,0 +1,231 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores command-line flags, environment variables, and configuration files used 
```

Comment:
It would be good though to have an example of what a ConfigData storing env, flags, and files might look like.

```
{
  "data": {
    "MY_ENV": "some var"
    ...
  }
}
```

etc